catroid/src/main/java/org/catrobat/catroid/formulaeditor/UserVariable.java

```diff
@@ -105,4 +106,18 @@ public boolean equals(Object obj) {
 	public int hashCode() {
 		return name.hashCode();
 	}
+
+	public String getDeviceValueFileName() {
+		return deviceValueFileName;
+	}
+
+	public UserVariable copyDeviceValueFileName(UserVariable variable) {
```

Comment:
what is this? what are you trying to do here? 

```
varA.copyDeviceValueFileName(varB);

```
this returns the variable that you called the method on and sets `varB`'s `deviceValueFileName` on `varA.

so all you can do with it is:

```
varA.setDeviceValueFileName(varB.getDeviceValueFileName())
```

so basically this should have been a setter.